node/src/Transport.ts

```diff
@@ -385,6 +385,63 @@ export class Transport<Events extends TransportEvents = TransportEvents,
 		this.#observer.safeEmit('close');
 	}
 
+	/**
+	 * Just called in WebRtcTransports when closing the associated WebRtcServer.
+	 *
+	 * @private
+	 * @virtual
+	 */
+	mustClose(): void
```

Comment:
Comment:

`close()`, `routerClosed` and `mustClose()` methods are identical with the exception that `close` sends a `transport.close` request through the channel. There is place for some code reduction here. Ie:

All of three calling an internal private `clearEverything( )` that clears Producers, Consumers, etc, etc.

If `reason === undefined` then send the 'transport.closed' request. Otherwise the code is identical for the three cases.

Reply:
If you carefully compare `close()` and `routerClosed()` you'll see that they are not that similar. The former emits many more events. I'd prefer to refactor this topic (if needed) in the future. Right now it's true that we have two methods that do **mostly** the same (but not even the same, see how `Transport.routerClosed()` emits 'routerclose' event while `Transport.mustClose()` doesn't, however `WebRtcTransport.mustClose()` calls parent `Transport.mustClose()` and then emits 'webrtcserverclose'). In any other cases this is not even true. Honestly I think that trying to unify those code fragments into a single method that receives arguments would make things harded to read.

Reply:
Hm, I understand why we have `close` and why `routerClosed` has a distinct use case of transport being closed involuntarily due to router above it closing. Should this one be called `webRtcServerClosed` then?

Reply:
We cannot have a method called `webRtcServerClosed()` in the `Transport` parent class :(

I'll explain it better below in response to your feedback.